src/components/dialogs/filters.js

```diff
@@ -83,6 +83,62 @@ export const CountriesSelection = ({
     );
 };
 
+export const EnumSelection = ({
+    initialValue,
+    onChange,
+    titleMessage,
+    enumValues,
+}) => {
+    const [value, setValue] = useState(initialValue);
+    const intl = useIntl();
+
+    const getEnumTranslation = useCallback(
```

Comment:
I would split the translation in two phases.
Cache the enumEntry -> localizedLabel in a plain js object,
then use direct it directly in the function passed to the autocomplete.
I would also cache the options given to the autoComplete.
With the simplification proposed in a comment well below this one, it would give smth like :
```
const enumLabels = useMemo(() => Object.fromEntries(
   Object.entries(enumValues)
     .map([k,v] => [k, intl.formatMessage({id:v})]),
   [intl, enumValues]);
const options = useMemo(() => Object.keys(enumValues), [enumValues]);
const getEnumTranslation = useCallback((enumEntry) => enumLabels[enumEntry], [enumLabels] ?? '');
```

Reply:
that's a pleasant refactor. I take it.